x/bank/client/cli/tx.go

```diff
@@ -27,6 +32,7 @@ func NewTxCmd() *cobra.Command {
 	txCmd.AddCommand(
 		NewSendTxCmd(),
 		NewMultiSendTxCmd(),
+		GetUpdateDenomMetadataProposal(),
```

Comment:
I'm not sure if this is the right place for this command. I'm not exactly sure what they're doing for stuff like this anymore. It used to be that these commands would get added to a special tx gov list, but with the recent x/gov changes, all of those commands are now listed under the `tx gov submit-legacy-proposal` command.

For an example see the `NewCmdSubmitLegacyUpgradeProposal` function and how it gets wired in.

Reply:
Actually, after looking at the syntax of the `update-denom-metadata-proposal` command, you can probably just leave it out completely. It's pretty much the same thing provided by the new `tx gov submit-proposal` command:

```
> build/provenanced tx gov submit-proposal --help
Submit a proposal along with some messages, metadata and deposit.
They should be defined in a JSON file.

Example:
$ provenanced tx gov submit-proposal path/to/proposal.json

Where proposal.json contains:

{
  // array of proto-JSON-encoded sdk.Msgs
  "messages": [
    {
      "@type": "/cosmos.bank.v1beta1.MsgSend",
      "from_address": "cosmos1...",
      "to_address": "cosmos1...",
      "amount":[{"denom": "stake","amount": "10"}]
    }
  ],
  "metadata: "4pIMOgIGx1vZGU=", // base64-encoded metadata
  "deposit": "10stake"
}

Usage:
  provenanced tx gov submit-proposal [path/to/proposal.json] [flags]

Flags:
  -a, --account-number uint      The account number of the signing account (offline mode only)
      --aux                      Generate aux signer data instead of sending a tx
  -b, --broadcast-mode string    Transaction broadcasting mode (sync|async|block) (default "sync")
      --dry-run                  ignore the --gas flag and perform a simulation of a transaction, but don't broadcast it (when enabled, the local Keybase is not accessible)
      --fee-granter string       Fee granter grants fees for the transaction
      --fee-payer string         Fee payer pays fees for the transaction instead of deducting from the signer
      --fees string              Fees to pay along with transaction; eg: 10uatom
      --from string              Name or address of private key with which to sign
      --gas string               gas limit to set per-transaction; set to "auto" to calculate sufficient gas automatically. Note: "auto" option doesn't always report accurate results. Set a valid coin value to adjust the result. Can be used instead of "fees". (default 200000)
      --gas-adjustment float     adjustment factor to be multiplied against the estimate returned by the tx simulation; if the gas limit is set manually this flag is ignored  (default 1)
      --gas-prices string        Gas prices in decimal format to determine the transaction fee (e.g. 0.1uatom)
      --generate-only            Build an unsigned transaction and write it to STDOUT (when enabled, the local Keybase only accessed when providing a key name)
  -h, --help                     help for submit-proposal
      --keyring-backend string   Select keyring's backend (os|file|kwallet|pass|test|memory) (default "test")
      --keyring-dir string       The client Keyring directory; if omitted, the default 'home' directory will be used
      --ledger                   Use a connected Ledger device
      --node string              <host>:<port> to tendermint rpc interface for this chain (default "tcp://localhost:26657")
      --note string              Note to add a description to the transaction (previously --memo)
      --offline                  Offline mode (does not allow any online functionality)
  -o, --output string            Output format (text|json) (default "json")
  -s, --sequence uint            The sequence number of the signing account (offline mode only)
      --sign-mode string         Choose sign mode (direct|amino-json|direct-aux), this is an advanced feature
      --timeout-height uint      Set a block timeout height to prevent the tx from being committed past a certain height
      --tip string               Tip is the amount that is going to be transferred to the fee payer on the target chain. This flag is only valid when used with --aux, and is ignored if the target chain didn't enable the TipDecorator
  -y, --yes                      Skip tx broadcasting prompt confirmation

Global Flags:
      --chain-id string          The network chain ID
      --custom-denom string      Indicates if a custom denom is to be used, and the name of it (default nhash)
      --home string              directory for config and data (default "/Users/danielwedul/git/provenance/build/run/provenanced")
      --log_format string        The logging format (json|plain) (default "plain")
      --log_level string         The logging level (trace|debug|info|warn|error|fatal|panic) (default "info")
      --msgfee-floor-price int   Custom msgfee floor price, optional (default 1905)
  -t, --testnet                  Indicates this command should use the testnet configuration (default: false [mainnet])
      --trace                    print out full stack trace on errors
```